pkg/reconciler/v1alpha1/taskrun/validate.go

```diff
@@ -24,14 +24,17 @@ import (
 
 // validate all references in taskrun exist at runtime
 func validateTaskRun(c *Reconciler, tr *v1alpha1.TaskRun) error {
-	// verify task reference exists, all params are provided
-	// and all inputs/outputs are bound
-	t, err := c.taskLister.Tasks(tr.Namespace).Get(tr.Spec.TaskRef.Name)
-	if err != nil {
-		return fmt.Errorf("Error listing task ref %s: %v",
-			tr.Spec.TaskRef.Name, err)
+	if tr.Spec.TaskRef != nil {
+		// verify task reference exists, all params are provided
+		// and all inputs/outputs are bound
+		t, err := c.taskLister.Tasks(tr.Namespace).Get(tr.Spec.TaskRef.Name)
+		if err != nil {
+			return fmt.Errorf("Error listing task ref %s: %v",
+				tr.Spec.TaskRef.Name, err)
+		}
+		return validateTaskRunAndTask(c, *tr, t, tr.Namespace)
 	}
-	return validateTaskRunAndTask(c, *tr, t, tr.Namespace)
+	return nil
```

Comment:
Shouldn't we validate it in that case too ? :thinking: 

This would require to change the signature of  `validateTaskRunTask` though, but most likely for the good

```go
// […]
    return validateTaskRunAndTask(c, *tr, t.Spec, tr.Namespace)
}
return validateTaskRunAndTask(c, *tr, tr.TaskSpec, tr.Namespace)
``

(On that note, thinking outloud, we *only* validate the spec, so we may want to pass only the specs — the `tr.Name` is only required on the error message, so we could use `github.com/pkg/errors` to wrap errors and thus adding the `tr.Name` via the caller)